prototypes/ScheduledMerges.hs

```diff
@@ -222,15 +225,20 @@ invariant = go 1
             -- same reasons as above. Although if this is the first merge for
             -- a new level, it'll have only 4 runs.
             (_, OngoingMerge _ rs _) -> do
-              assertST $ length rs == 4 || length rs == 5
-              assertST $ all (\r -> tieringRunSizeToLevel r `elem` [ln-1, ln]) (take 4 rs)
-              assertST $ all (\r -> levellingRunSizeToLevel r <= ln+1) (drop 4 rs)
+              let incoming = take 4 rs
+              let resident = drop 4 rs
+              assertST $ length incoming == 4
+              assertST $ length resident <= 1
```

Comment:
Ok, ok, this is indeed a funny but correct way of saying that length rs is 4 or 5. "the bit less than or equal to 4 must be exactly 4 and the bit more than 4 must be at most 1". :exploding_head: 

I don't think it's clearer!

Reply:
I think it helps in understanding why those two cases are possible. But since the comment above explains it as well, I can change it to ``length rs `elem` [4, 5]``. :+1: 
The important part is that `incoming` and `resident` have different invariants in the checks in the following two lines.